src/Control/Carrier/Diagnostics.hs

```diff
@@ -67,6 +67,9 @@ data ResultBundle a = ResultBundle
     resultValue :: a
   }
 
+instance (Show a) => Show (ResultBundle a) where
+  show = show . resultValue
```

Comment:
We are not using this method but Connor and I ran into this issue when debugging so we thought it would be best to implement this function for anyone in the future who realized that ResultBundle doesn't have an instance of `Show`. The goal of this was to show the result as easily as possible, but I do understand your concern about not showing the warnings. @cnr I think your opinion would be helpful here. I think I'd like to do something like concatenating the result and warnings.

Reply:
I'd prefer to just derive it on `ResultBundle` and `SomeDiagnostic`.  That should handle it enough for our debugging purposes.  We can decide a solid long term solution later.  If you can't derive it for `ResultBundle`, then this function should just `show` the list of warnings, then `show` the `resultValue`.